workflow/common/util.go

```diff
@@ -100,7 +101,7 @@ func ExecPodContainer(restConfig *rest.Config, namespace string, pod string, con
 func GetExecutorOutput(exec remotecommand.Executor) (*bytes.Buffer, *bytes.Buffer, error) {
 	var stdOut bytes.Buffer
 	var stdErr bytes.Buffer
-	err := exec.Stream(remotecommand.StreamOptions{
+	err := exec.StreamWithContext(context.TODO(), remotecommand.StreamOptions{
```

Comment:
this would need a bit of refactoring for the entire function chain to pass a `context` down, but this could inherit the `context` of [`processNextPodCleanupItem`](https://github.com/argoproj/argo-workflows/blob/748ae475d3400f2aa51800c2337607811550d989/workflow/controller/controller.go#L546)

Reply:
Neat, I've opened https://github.com/argoproj/argo-workflows/pull/12858